Source/WebCore/platform/graphics/skia/ComplexTextControllerSkia.cpp

```diff
@@ -191,7 +192,15 @@ void ComplexTextController::collectComplexTextRunsForCharacters(std::span<const
     }
 
     HbUniquePtr<hb_buffer_t> buffer(hb_buffer_create());
+
     for (unsigned i = 0; i < runCount; ++i) {
+        // The computed "locale" equals the "lang" attribute. The latter must be a valid BCP 47 language tag,
+        // according to <https://html.spec.whatwg.org/multipage/dom.html#attr-lang>.
+        // According to <https://datatracker.ietf.org/doc/html/rfc5646#section-2.1>
+        // "the language tags described in this document are sequences of characters
+        // from the US-ASCII [ISO646] repertoire.".
+        hb_buffer_set_language(buffer.get(), hb_language_from_string(m_font.fontDescription().computedLocale().string().ascii().data(), -1));
```

Comment:
The language is the same for all runs, so I would get the hb_language_t once before the loop and use that value in every loop iteration.